agent/acs/handler/acs_handler.go

```diff
@@ -435,9 +433,15 @@ func newDisconnectionTimer(client wsclient.ClientServer, timeout time.Duration,
 
 // anyMessageHandler handles any server message. Any server message means the
 // connection is active and thus the heartbeat disconnect should not occur
-func anyMessageHandler(timer ttime.Timer) func(interface{}) {
+func anyMessageHandler(timer ttime.Timer, client wsclient.ClientServer) func(interface{}) {
 	return func(interface{}) {
-		seelog.Debug("ACS activity occured")
+		seelog.Debug("ACS activity occurred")
+		// Reset read deadline as there's activity on the channel
+		if err := client.SetReadDeadline(time.Now().Add(heartbeatTimeout + heartbeatJitter)); err != nil {
```

Comment:
Can you explain again here too? Doesn't doubling that have the net effect of doubling your pre-established `heartbeatTimeout` the first time there is some traffic over the websocket?

Reply:
Because we use [`rand.Int63n()` in `AddJitter`](https://github.com/aws/amazon-ecs-agent/blob/master/agent/utils/backoff.go#L76), `heartbeatTimeout + heartbeatJitter` should always be greater than the heartbeat duration that we compute. I think it should be safe to set it the way it's currently being set.

Reply:
>  Doesn't doubling that have the net effect of doubling your pre-established `heartbeatTimeout` the first time there is some traffic over the websocket?

No, the deadline is set based on `time.Now()`, so it's always in the future by the amount specified here.

> I think it should be safe to set it the way it's currently being set.

I'm worried, because this is a time and not a duration, and because Go 1.7 doesn't have a monotonic clock, that something unexpected can happen here.  I'd feel more comfortable with a longer timeout because we don't expect this timeout to be hit anyway and I don't see any danger of making it longer.

Once we move to Go 1.9 there will be a monotonic clock and this should be safer.

Reply:
> I'd feel more comfortable with a longer timeout because we don't expect this timeout to be hit anyway

Oh, you'd be surprised :) I think we'll hit this timeout when there's connection issues. However, I do buy the argument that `2*heartbeatTimeout + heartbeatJitter` will work for most cases here.